R/make-traceability-matrix.R

```diff
@@ -0,0 +1,83 @@
+
+#' Build the Traceability Matrix and write it to output file(s)
+#' @importFrom purrr map walk
+#' @importFrom dplyr arrange first group_by select
+#' @importFrom knitr kable
+#' @importFrom tidyr unnest
+#' @importFrom glue glue
+#' @importFrom rmarkdown render
+#' @importFrom rlang .data
+#' @importFrom stringr str_extract str_trim
+#' @importFrom fs dir_exists dir_create
+#' @param df Tibble containing stories, requirements, and tests. Created in
+#'   [create_validation_docs()].
+#' @param product The name of the product you are validating, to be included in the output document.
+#' @param version The version number of the product you are validating, to be included in the output document.
+#' @param style_dir Directory to check for a docx style reference that has the
+#'   same base name as `out_file`.
+#' @param out_file filename to write markdown file out to. Any extension will be ignored and replaced with .md
+#' @param output_dir Directory to write the output documents to. Defaults to working directory.
+#' @param word_document Logical scaler indicating whether to render a docx document
+#' @keywords internal
+make_traceability_matrix <- function(
+  df,
+  product,
+  version,
+  style_dir = NULL,
+  out_file = MAT_FILE,
+  output_dir = getwd(),
+  word_document = TRUE
+) {
+
+  template <- get_template("traceability_matrix", type = type)
+
+  if (!fs::dir_exists(output_dir)) fs::dir_create(output_dir)
+  out_file <- file.path(output_dir, paste0(tools::file_path_sans_ext(out_file), ".Rmd"))
+  fs::file_copy(template, out_file)
+
+  mat <- df %>%
+    filter(!is.na(.data$StoryId)) %>%
+    unnest(cols = c(.data$tests)) %>%
+    filter(!is.na(.data$passed))
+
+  mat <- if ("RequirementId" %in% names(mat)) {
+    arrange(mat, .data$StoryId, .data$RequirementId, .data$TestId)
+  } else {
+    arrange(mat, .data$StoryId, .data$TestId)
+  }
+
+  mat <- group_by(mat, .data$StoryId) %>%
+    summarise(test_ids = paste0(.data$TestId, collapse = " "),
+              description = first(.data$StoryDescription)) %>%
+    mutate(description = str_extract(str_trim(.data$description), "^.+"))
+
```

Comment:
I'm not exactly sure where in the code it goes, but if you look at the new template, they want the story ID in with the story description:
<img width="676" alt="Screen Shot 2022-08-03 at 1 36 59 PM" src="https://user-images.githubusercontent.com/7806083/182673397-16296806-6aae-44d5-a3d5-31cb55e65c56.png">

I think maybe put it in parantheses in that same column. So that the results would look something like this:

<img width="696" alt="Screen Shot 2022-08-03 at 1 40 31 PM" src="https://user-images.githubusercontent.com/7806083/182673805-5412be43-8d3c-43fa-a71f-dd6fd168c2ca.png">

Reply:
Ahh good catch. Resolving in next commit